iwyu_test_util.py

```diff
@@ -73,12 +73,54 @@ class Features:
   # Parses the output of 'include-what-you-use -print-targets'.
   SUPPORTED_TARGET_RE = re.compile(r'^\s+([a-z0-9_-]+)\s* - .*$')
 
+  # Parses the output of 'include-what-you-use -dM'.
+  PREPROCESSOR_MACRO_RE = re.compile(r'^#define ([A-z_][A-z0-9_]*)[\s\(].*$')
+
   def __init__(self):
     # Maps all known features names to prerequisite predicates.
     self.known = {
-      'has-target': Features.HasTarget
+      'has-target': Features.HasTarget,
+      'using-cstdlib': Features.UsingCstdlib,
+      'using-cxxstdlib': Features.UsingCXXstdlib,
     }
 
+  def _GetPreprocessorMacros(lang, header):
+    iwyu = _GetIwyuPath()
+    txt = f'#include {header}'
+
+    result = subprocess.run([iwyu, '-dM', '-x', lang, '-'], capture_output=True,
+                            input=txt, text=True)
+
+    macros = []
+    for line in result.stdout.splitlines():
+      m = Features.PREPROCESSOR_MACRO_RE.match(line)
+      if m:
+        macros.append(m.group(1))
+
+    return macros
+
+  # TODO: cache result
+  @staticmethod
+  def _GetCXXPreprocessorMacros():
+    print("\n".join(Features._GetPreprocessorMacros('c++', '<cstddef>')))
+    return Features._GetPreprocessorMacros('c++', '<cstddef>')
+
+  # TODO: cache result
+  @staticmethod
+  def _GetCPreprocessorMacros():
+    print("\n".join(Features._GetPreprocessorMacros('c', '<stddef.h>')))
+    return Features._GetPreprocessorMacros('c', '<stddef.h>')
+
+  @staticmethod
+  def _CheckForCXXPreprocessorMacro(macro):
+    """ Look for the specified macro in the C++ preprocessor output. """
+    return macro in Features._GetCXXPreprocessorMacros()
+
+  @staticmethod
+  def _CheckForCPreprocessorMacro(macro):
+    """ Look for the specified macro in the C preprocessor output. """
+    return macro in Features._GetCPreprocessorMacros()
```

Comment:
I don't think these two methods are necessary, let's just inline the `macro in` into the `Using.*Lib` methods?